Eastern-Research-Group/expert-query
# Feature/130 download button

## Related Issues:
* [EQ-130](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-130)

## Main Changes:
* Added functionality to the _Download_ button in the client interface.
* Added a modal that displays the current query's row count before executing the query and downloading data.
    * The OOTB **USWDS** modal component did not work as expected, so a custom `Modal` component was created, based heavily on the homonymous component from the [react-uswds](https://github.com/trussworks/react-uswds) repository. This component has more features than we need: they can be left in or further trimmed down.
* Added a missing search icon to the static **HTML** pages.
* Removed the `uswds.js` import from `app.tsx`, since it was interfering with the EPA template.

## Steps To Test:
1. Navigate to http://localhost:3000/attains#dataProfile=assessmentUnits&format=xlsx&assessmentUnitId=AL03130002-0804-400.
2. Click the _Download_ button, then click _Continue_ in the modal that appears.
3. Verify that the file download process proceeds as expected, and that the file contents are correct.
4. Build a query for a profile that is not yet connected, e.g. http://localhost:3000/attains#dataProfile=assessments&format=xlsx&assessmentUnitId=AL03130002-0804-400.
5. Click the _Download_ button, and confirm that an error message is displayed in a modal.
6. Repeat as needed with different profiles, filters, and file formats.

## TODO:
- [ ] We'll need to establish a row count cutoff before sending users to the national downloads, as well as the UI behavior once that cutoff is exceeded.
